### PR TITLE
Use multiQueryUpdater in update2 to properly sort/filter

### PR DESCRIPTION
**Pull request by yannick-mayeur**
This needs to be properly tested before getting merged. Based on Unit
Tests it does not break anyting.

When building the demo application for the new datatable with @loustak,
we will make sure to try this commit out.

### CODE REVIEW

**Comment by yannick-mayeur** [comment]
Maybe this turns out to be useless thanks to https://github.com/VulcanJS/Vulcan/pull/2494

File: packages/vulcan-core/lib/modules/containers/update2.js [hunk 1]
```diff
@@ -58,6 +60,7 @@ export const useUpdate2 = (options) => {
   const [updateFunc, ...rest] = useMutation(query, {
     // see https://www.apollographql.com/docs/react/features/error-handling/#error-policies
     errorPolicy: 'all',
+    update: multiQueryUpdater({ typeName, fragment, fragmentName, collection, resolverName: `update{typeName}` }),
```
Comment (eric-burel) [inline packages/vulcan-core/lib/modules/containers/update2.js]: it's `update${typeName}` you missed a $

**Comment by eric-burel** [comment]
As far as I understand #2494 is targeted at custom update functions, so it is compatible with what you've done
I've just added a comment because the resolver name was wrong

**Comment by eric-burel** [comment]
It sounds fine to me, let me know when you think it's ok to merge. Thanks!

**Comment by yannick-mayeur** [comment]
I finally got around to testing this.
In the `getting-started` package the `MovieApp2` component uses the `Datatable` component, which I think uses `multi2`.
But when updating it first stays at position it was and then moves to the correct one. The same thing happens when creating a new object though (it first appears at the top of the list and then after a few seconds moves to the correct position).

**Comment by eric-burel** [comment]
Ok can you mark this PR as valid (not a draft, I can't do it myself)? I'll merge it because at least update is consistent with create.

**Comment by eric-burel** [comment]
@yannick-mayeur Datatable uses the new multi but the form itself do not use the new `create`/`update` so I think it makes sense. The `multiQueryUpdater` is tied to the mutation create/update, not to the multi query.

**Comment by eric-burel** [comment]
Yeah I confirm I was wrong in https://github.com/VulcanJS/Vulcan-Starter/issues/140, the issue is not the DataTable using multi2, it's the Smart form using create/update.
So basically we probably need to update the smart form instead. But the behaviour is consistent, I think the multiQueryUpdater is ok in itself. Can you please also confirm #2494  is ok to be merged?

**Comment by yannick-mayeur** [comment]
https://github.com/VulcanJS/Vulcan/pull/2494 is closed

**Comment by eric-burel** [comment]
Yes I mean the newer one https://github.com/VulcanJS/Vulcan/pull/2501

**Comment by yannick-mayeur** [comment]
With #2501, when creating the item does not appear for few seconds but when it appears it directly at the correct position.

**Comment by eric-burel** [comment]
That's not good it should appear right away. When it appears only after a few seconds, it's because of a polling request (you can see it in the Network tab). The multiQueryUpdater should make it appear immediately (ideally at the correct position but that's the another issue).

**Comment by eric-burel** [comment]
@yannick-mayeur I've fixed the order issue. The solution is to force a default order in the default resolver. Otherwise you can't tell in which order the data will be returned by the server, so it's impossible to match the order client side and server side. With a default order, if no `sort` is provided, it will now use `{createdAt: -1}` so the order stays consistent.

Please mark your PR as not a draft so I can merge it. Thanks a lot for investigating this.

**Comment by yannick-mayeur** [comment]
Great!

I marked the PR ready for review.

**Pull request merged**